rotkehlchen/serialization/deserialize.py

```diff
@@ -578,10 +607,36 @@ def deserialize_evm_transaction(
             gas_used = read_integer(data, 'gasUsed', source)
         nonce = read_integer(data, 'nonce', source)
 
-        if chain_id in L2_CHAINIDS_WITH_L1_FEES and evm_inquirer is not None:
-            if not raw_receipt_data:
-                raw_receipt_data = evm_inquirer.get_transaction_receipt(tx_hash)
-            l1_fee = maybe_read_integer(raw_receipt_data, 'l1Fee', source)
+        if chain_id in L2_CHAINIDS_WITH_L1_FEES:
+            if (
+                (l1_fee := data.get('L1FeesPaid')) is None and  # etherscan txlist uses this key
+                evm_inquirer is not None
+            ):
+                if raw_receipt_data is None:
+                    raw_receipt_data = _get_transaction_receipt(
+                        tx_hash=tx_hash,
+                        chain_id=chain_id,
+                        timestamp=timestamp,
+                        evm_inquirer=evm_inquirer,
+                    )
+                try:
+                    l1_fee = maybe_read_integer(raw_receipt_data, 'l1Fee', source)
+                except DeserializationError as e:  # Fall back to etherscan (via txlist)
+                    log.warning(f'Failed to get L1 fee from receipt due to {e!s}. Falling back to etherscan.')  # noqa: E501
+                    l1_fee = evm_inquirer.etherscan.maybe_get_l1_fees(
```

Comment:
I do not understand why you need this logic and this extra function. Don't we call txlist when we originally get the transactions? So shouldn't the l1fee be there?